src/components/CharacterBox/CharacterBox.js

```diff
@@ -10,50 +12,52 @@ export default class CharacterBox extends Component {
     }
   }
 
-  componentDidMount = () => {
-    const limitedCharacters = this.props.characters.slice(0, 10)
-    const characterPromises = limitedCharacters.map(character => {
-      return fetch(character)
-      .then(response => response.json())
-      .catch(err => {
-        this.setState({characters: [...this.state.characters, character]})
-      })
-    })
-
-    return Promise.all(characterPromises)
-      .then(promises => promises.map(promise => {
+  componentDidMount() {
+    return this.addCharacters()
+  }
 
-        const characterName = promise.name;
+  speciesFetch = (url) => {
+    return getAnyData(url, 'species')
+      .then(data => {return {species: data.name}})
+  }
 
-        const speciesFetch =
-        fetch(promise.species)
-        .then(res => res.json())
-        .then(data => data.name)
+  planetFetch = (url) => {
+    return getAnyData(url, 'planet')
+      .then(data => {return {planet: data.name, population: data.population}})
+  }
 
-        const planetFetch =
-        fetch(promise.homeworld)
-        .then(res => res.json())
-        .then(data => {return {planetName: data.name, planetPopulation: data.population}})
+  filmsFetch = (url) => {
+    return Promise.all(url.map((film, index) => {
+      return getAnyData(film, `accociated film ${index + 1}`)
+        .then(data => data.title)}))
+        .then(data => {return {films: data}})
+    }
 
-        const filmsPromises =
-        promise.films.map(film => {
-          return fetch(film)
-          .then(res => res.json())
-          .then(data => data.title)
+  nestedCharacterFetch = (promise) => {
+    const name = promise.name;
+    return Promise.all([this.speciesFetch(promise.species), this.planetFetch(promise.homeworld), this.filmsFetch(promise.films)])
+      .then(info => {
+        return {name, ...info[0], ...info[1], ...info[2]}
+      })
+        .then(characterStats => {
+          this.setState({characters: [...this.state.characters, characterStats]})
         })
+  }
```

Comment:
I made small changes to this function. I guess it looks cleaner

<img width="715" alt="Screen Shot 2019-12-08 at 8 52 39 PM" src="https://user-images.githubusercontent.com/48605821/70406413-dccc0580-19fd-11ea-853e-bf5fa5a555a7.png">

Reply:
ooh nice destructuring!